python/pyspark/sql/types.py

```diff
@@ -1679,6 +1678,15 @@ def from_arrow_schema(arrow_schema):
          for field in arrow_schema])
 
 
+def _require_minimum_pyarrow_version():
```

Comment:
@ueshin @HyukjinKwon just want to verify that we want to check for a minimum version of pyarrow and this is the right place to put this function?

Reply:
I'm not familiar with `LooseVersion`, but don't we need to use `LooseVersion` for `pyarrow.__version__`, too? cc @HyukjinKwon 
Btw, shall we add the dependency to `setup.py`?

Reply:
Seems fine. Only I know about `LooseVersion` is it compares versions in string correctly. Sure, we should add it there too.

Reply:
>  don't we need to use LooseVersion for pyarrow.__version__, too?

Seems fine by 

https://github.com/python/cpython/blob/6f0eb93183519024cb360162bdd81b9faec97ba6/Lib/distutils/version.py#L331-L340

Reply:
> just want to verify that we want to check for a minimum version of pyarrow and this is the right place to put this function?

Could we put this under `pyspark.sql.utils` for now if we are all fine?

Reply:
sounds good